fem/intrules.hpp

```diff
@@ -26,8 +26,12 @@ class IntegrationPoint
 {
 public:
    double x, y, z, weight;
+   int index;
```

Comment:
I thought about this modification and I don't see a nicer way of providing the information you need without changing the current API.

However, anyone can change `index` being a public member. What about making it `const` and private? And have an API to access it?

Reply:
Here is my take on why it is OK to keep these public.

Typically when `IntegrationRule` is obtained from an `IntegrationRules` container (e.g. the global `IntRules` container), it returns a `const IntegrationRule &`:
https://github.com/mfem/mfem/blob/248debb3dc109f36833a83e9f660225c5b676ec4/fem/intrules.hpp#L356-L357

From the `const IntegrationRule &`, one can access the `IntgrationPoint`s only as `const IntegrationPoint &`, so none of its data members can be modified. This is one reason why we kept all fields public. In this regard, `index` is not different from the other fields: `x`, `y`, etc.

Reply:
ok, fine with me.